### ISSUE TITLE
the policy perform bad in my own env

### ISSUE THREAD

**Issue by wenzhoulyu** [lmzintgraf/varibad#9]
I'm using the code for my own env which has no time limit but has max episode step limit.
and my best action maybe about -0.5. But I found that my action would beyound the limitation  [-1,1] a lot. It can’t find a good policy.  I wonder how could I solve this problem. Thank you so much.
![397F807C-CB20-48A3-B94E-7E8788E10230](https://user-images.githubusercontent.com/47205275/161812840-d3d97759-fb03-4734-819e-404545b1bc03.jpeg)
![9F806708-E2A0-4AE0-BF6C-3076F66E4B10](https://user-images.githubusercontent.com/47205275/161812846-a2bd95a7-f614-4658-ba8e-6ad2765bc355.jpeg)
![92A7541A-C374-4F2B-B571-691B1A48FA6C](https://user-images.githubusercontent.com/47205275/161812850-57560a15-7d35-4a47-9aae-a85d484999ca.jpeg)

**Comment by lmzintgraf** [comment]
Hi, 

Thanks for your interest in our code. 

I have a few questions
- Can you clarify what the difference between `time limit` and `max episode step limit` is?
- What type of env is it (e.g., continuous or discrete states/actions)?
- Which config file did you build on? 
- Did you double check that your environments don't use `np.random` anywhere?
- Did you run the meta-learning code, or did you just train a single policy? (See my first suggestion below)

Here are a few things I recommend you try: 
- First train an "expert", so see if your baseline works: run PPO on a _single_ environment, without any meta-learning. You can build on the config files that end with `_expert`. This should work before you do anything else, and is a good way to tune the PPO hyper-parameters.
- Try to find out whether it's the VAE or the policy that's not learning (e.g., by analysing the loss curves or looking at what the VAE predicts) - that might be a good starting point to debug. 
- Normalising actions in [-1, 1] can be tricky; did you try setting either `norm_actions_pre_sampling` or `norm_actions_post_sampling` to True? This sometimes helps.

**Comment by wenzhoulyu** [comment]
Well In my env, max episode step limit means that I don't want the env last too long (it doesn't mean the observation is bad). If i don't add the max_episode_step limit, i think the env maybe never done in a good policy. I don't konw whether it will affect the algorithm.  Because I find that in the `storage_vae.py`, if the task ended you will dump the data into the larger buffer.
My env don't have the np.random except the _seed(), but i find your code also has the np.random in _seed()
I'll try to run PPO on a single environment, thank you so much

**Comment by lmzintgraf** [comment]
I see, that's similar to my gridworld experiments and some of the mujoco tasks, so it shouldn't be a problem (just double check that it actually returns `done` when the time limit is reached - the data should be dumped into the larger VAE buffer at that point). You can try setting `use_proper_time_limits` to True, to treat the timeouts correctly, but this shouldn't make or break the entire algorithm. 

On the seed thing, just to make sure we don't misunderstand each other: the code in general can have `np.random`, but _not_ the environment itself. This is because multiple environments are instantiated and ran in parallel, and `np.random` is not thread-safe. 

Good luck! Closing this for now, but feel free to re-open if you have more questions.
